### ISSUE TITLE
IBM null_args_for_empty_datatypes test failing

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#1674]
@bosilca We're seeing the new IBM test fail (https://mtt.open-mpi.org/index.php?do_redir=2320)

```
MPI_Type_create_darray[mpi003:14183] *** An error occurred in MPI_Type_create_darray
[mpi003:14183] *** reported by process [2437611521,10]
[mpi003:14183] *** on communicator MPI_COMM_WORLD
[mpi003:14183] *** MPI_ERR_ARG: invalid argument of some other kind
[mpi003:14183] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[mpi003:14183] ***    and potentially your MPI job)MPI_Type_contiguous
MPI_Type_create_darrayMPI_Type_contiguous
MPI_Type_create_darray[mpi003:14154] 31 more processes have sent help message help-mpi-errors.txt /
mpi_errors_are_fatal
[mpi003:14154] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
```

Can you have a look?

**Comment by bosilca** [comment]
open-mpi/ompi@6e6ed62a has not bee cherry-picked in 2.x.

**Comment by jsquyres** [comment]
Fixed via open-mpi/ompi-release#1168.
